dask/dataframe/tests/test_dataframe.py

```diff
@@ -709,6 +709,17 @@ def test_size():
     assert_eq(d.index.size, full.index.size)
 
 
+def test_shape():
+    result = d.shape[0].compute()
+    assert_eq(result, len(full))
+
+    result = d.shape[1].compute()
+    assert_eq(result, len(full.columns))
+
+    result = d.a.shape[0].compute()
+    assert_eq(result, len(full.a))
```

Comment:
Would it work to call `assert_eq` directly on the delayed object like `assert_eq(d.shape, full.shape)`?

The assert_eq function also checks a few things on task graphs that we probably want to keep if possible

Reply:
I believe this is not possible. At least I got this message:

> TypeError: Truth of Delayed objects is not supported